src/dune_engine/build_system.ml

```diff
@@ -476,10 +482,12 @@ end = struct
           in
           let produced_targets =
             match sandbox with
-            | None ->
+            | None -> (
               (* Directory targets are not allowed for non-sandboxed actions, so
                  the call below should not raise. *)
```

Comment:
That's a stale comment, right? The above change allows directory targets for some non-sandboxed actions.